### PR TITLE
move Build.Status.Config ref to Build.Spec

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
[testextended][extended:core(builds)]

**Comment by bparees** [comment]
[testextended][extended:core(images)]

**Comment by bparees** [comment]
@csrwng @smarterclayton ptal.  not for 3.3, but soon after.  this is one of the necessary steps to get to the point where build.Spec can be treated as immutable and no build input information comes out of build.Status.

File: api/swagger-spec/oapi-v1.json [hunk 1]
```diff
@@ -22067,7 +22071,7 @@
      },
      "config": {
       "$ref": "v1.ObjectReference",
-      "description": "config is an ObjectReference to the BuildConfig this Build is based on."
+      "description": "config is an ObjectReference to the BuildConfig this Build is based on. This field is deprecated in deference to Spec.Config."
```
Comment (smarterclayton) [inline api/swagger-spec/oapi-v1.json]: You can't make spec.config authoritative, that is not backwards compatible for a client that doesn't know about spec.config.  

I don't see a way to add this to v1 without it being a breaking change.

Comment (bparees) [inline api/swagger-spec/oapi-v1.json]: the conversion still respects status if it's provided, though....  spec is only used if it's provided and non-nil, isn't that sufficient to be non-breaking for older clients?

**Pull request closed without merging**